R/estimate_dispersion.R

```diff
@@ -103,40 +125,69 @@ estimate_dispersion <- function(
   }
 
 
-  n_horizons <- ncol(list_of_ncs[[1]]) - 1
+  n_possible_horizons <- ncol(list_of_ncs[[1]]) - 1
   # Each row is retrospective nowcast date, each column is a horizon (i.e
   # columns are not delays, but hoirzons, and each cell contains a total
   # value corresponding to that horizon -- the total expected value to add
   exp_to_add <-
-    to_add_already_observed <- matrix(NA, nrow = n, ncol = n_horizons)
+    to_add_already_observed <- matrix(NA, nrow = n, ncol = n_possible_horizons)
   for (i in seq_len(n)) { # Seq along retrospective forecast dates
     # Rretrospective nowcast as of i delays ago
     nowcast_i <- list_of_ncs[[i]]
     # Remove the last i observations
     trunc_matr_observed <- list_of_obs[[i]]
     max_t <- nrow(trunc_matr_observed)
+    n_horizons <- min(max_t - k + 1, n_possible_horizons)
+    if (i == 1 && n_horizons < n_possible_horizons) {
+      cli_abort(
+        message = c(
+          sprintf(
+            "Requested window size k=%i is too large to generate sufficient nowcasts for the required forecast horizons for all available nowcast matrices.", # nolint
```

Comment:
For the future rather than just using no lint here it would be better to deal with the linting issue